pkg/cmd/cli/config/store.go

```
@@ -0,0 +1,185 @@
+		}
+	}
+
+	return configStore, created, err
+}
+
+// Save the provided config attributes to this ConfigStore.
+func (c *ConfigStore) SaveToFile(username, token string, namespace string, clientCfg *client.Config, rawCfg clientcmdapi.Config) error {
+	glog.V(4).Infof("Trying to merge and update %v config to '%v'...", c.providerEngine, c.Path)
+
+	config := clientcmdapi.NewConfig()
+
+	credentialsName := username
+	if len(credentialsName) == 0 {
+		credentialsName = "osc-login"
+	}
+	credentials := clientcmdapi.NewAuthInfo()
+	credentials.Token = token
+	config.AuthInfos[credentialsName] = *credentials
+
+	serverAddr := flagtypes.Addr{Value: clientCfg.Host}.Default()
+	clusterName := fmt.Sprintf("%v:%v", serverAddr.Host, serverAddr.Port)
+	cluster := clientcmdapi.NewCluster()
+	cluster.Server = clientCfg.Host
+	cluster.InsecureSkipTLSVerify = clientCfg.Insecure
+	cluster.CertificateAuthority = clientCfg.CAFile
+	config.Clusters[clusterName] = *cluster
+
+	contextName := clusterName + "-" + credentialsName
+	context := clientcmdapi.NewContext()
+	context.Cluster = clusterName
+	context.AuthInfo = credentialsName
+	context.Namespace = namespace
+	config.Contexts[contextName] = *context
+
+	config.CurrentContext = contextName
+
+	configToModify := c.Config
+
+	configToWrite, err := MergeConfig(rawCfg, *configToModify, *config)
+	if err != nil {
+		return err
+	}
+
+	// TODO need to handle file not writable (probably create a copy)
+	err = clientcmd.WriteToFile(*configToWrite, c.Path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func tryToLoad(path string, providerEngine string) (*ConfigStore, error) {
+	if len(path) > 0 {
+		config, err := clientcmd.LoadFromFile(path)
+		if err == nil {
+			glog.V(4).Infof("Using config from %v", path)
+			return &ConfigStore{config, path, providerEngine}, nil
+		} else {
+			glog.V(5).Infof("Unable to load config file for %v: %v", providerEngine, err.Error())
+			return nil, fmt.Errorf("Config file not found in %v", path)
```

Don't suppress the original errors.